File: nim/allergies/allergies.nim
type Allergies* = object
    score*: int
    
let foods = 
    ["eggs",
    "peanuts",
    "shellfish",
    "strawberries",
    "tomatoes",
    "chocolate",
    "pollen",
    "cats"]

proc isAllergicTo*(allergies: Allergies, food: string): bool =
    let lowscore = allergies.score mod 256
    let mask = 1 shl find(foods, food)
    return (lowscore and mask) != 0

proc lst*(allergies: Allergies): seq[string] =
    for food in foods:
        if allergies.isAllergicTo(food):
            result.add(food)

File: julia/allergies/allergies.jl
foods = 
    ["eggs",
    "peanuts",
    "shellfish",
    "strawberries",
    "tomatoes",
    "chocolate",
    "pollen",
    "cats"]

function allergic_to(score, allergen)
    lowscore = score % 256
    indexArray = findall( x -> x == allergen, foods )
    mask = 1 << (indexArray[1] - 1)
    (lowscore & mask) != 0
end

function allergy_list(score)
    filter(food -> allergic_to(score, food), foods) |> Set
end
